### PR TITLE
Added Kusto

### PR DESCRIPTION
**Pull request by oflipman**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by adewaleo** [comment]
have you setup a command review with the team @oflipman?

cc: @tjprescott

**Comment by oflipman** [comment]
Hi @adewaleo,
Thanks for the response.
I did not set up a command review, and I would like to set it please.
With whom should I set it with?

**Comment by adewaleo** [comment]
Feel free to email the team here **Azure Python CLI WG <AzPyCLI@microsoft.com>**

File: src/command_modules/azure-cli-kusto/HISTORY.rst [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.. :changelog:
+
+Release History
+===============
+
+0.1.0 (2016-1-4)
++++++++++++++++++++++
```
Comment (tjprescott) [inline src/command_modules/azure-cli-kusto/HISTORY.rst]: This should be
```
0.1.0
+++++
```

The number of + signs matters.

**Comment by oflipman** [comment]
@tjprescott.
All the test passed.
Can I ask you to review this PR?

File: azure-cli.pyproj [hunk 2]
```diff
@@ -12,7 +12,7 @@
     <OutputPath>.</OutputPath>
     <ProjectTypeGuids>{888888a0-9f3d-457c-b088-3a5042f75d52}</ProjectTypeGuids>
     <LaunchProvider>Standard Python launcher</LaunchProvider>
-    <InterpreterId>MSBuild|{f0ce7f81-ab4f-4088-b037-81002e6ddc44}|$(MSBuildProjectFullPath)</InterpreterId>
```
Comment (tjprescott) [inline azure-cli.pyproj]: Please revert this change.

File: azure-cli.pyproj [hunk 3]
```diff
@@ -1309,6 +1328,24 @@
       <PathEnvironmentVariable>PYTHONPATH</PathEnvironmentVariable>
       <Architecture>Amd64</Architecture>
     </Interpreter>
+    <Interpreter Include="..\env\">
```
Comment (tjprescott) [inline azure-cli.pyproj]: Please revert this change.

File: src/command_modules/azure-cli-kusto/azure/cli/command_modules/kusto/_params.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+def load_arguments(self, _):
+
+    # Kusto clusters
+    sku_arg_type = CLIArgumentType(help='The name of the sku.',
+                                   arg_type=get_enum_type(AzureSkuName))
+
+    with self.argument_context('kusto cluster') as c:
+        c.ignore('kusto_management_request_options')
+        c.argument('cluster_name', arg_type=name_type, help='The name of the cluster.', id_part='name')
+        c.argument('sku', arg_type=sku_arg_type)
```
Comment (tjprescott) [inline src/command_modules/azure-cli-kusto/azure/cli/command_modules/kusto/_params.py]: You also need `sku_name` to reference this arg_type so that it will be correctly applied to your `cluster update` command.

File: src/command_modules/azure-cli-kusto/azure/cli/command_modules/kusto/custom.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+                       cluster_name=cluster_name,
+                       database_name=database_name,
+                       parameters=_database,
+                       custom_headers=custom_headers,
+                       raw=raw,
+                       polling=polling,
+                       operation_config=kwargs)
+
+
+def update_kusto_cluster(instance, sku_name=None, capacity=None):
```
Comment (tjprescott) [inline src/command_modules/azure-cli-kusto/azure/cli/command_modules/kusto/custom.py]: Alternatively, just change `sku_name` here to `sku` and it will pick up the same metadata as the create command.

File: src/command_modules/azure-cli-kusto/azure/cli/command_modules/kusto/custom.py [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+    from azure.mgmt.kusto.models import AzureSku
+    if sku_name is None:
+        sku_name = instance.sku.name
+    if capacity is None:
+        capacity = instance.sku.capacity
+    instance.sku = AzureSku(name=sku_name, capacity=capacity)
+    return instance
+
+
+def update_kusto_database(instance, soft_delete_period_in_days, hot_cache_period_in_days=None):
```
Comment (tjprescott) [inline src/command_modules/azure-cli-kusto/azure/cli/command_modules/kusto/custom.py]: You need to alias these to `--soft-delete-period` and `--hot-cache-period` as we don't allow units in parameter names. You can either add the alias in params.py or change the variable names here accordingly.
